yt_dlp/extractor/joqrag.py

```diff
@@ -0,0 +1,54 @@
+import urllib.parse
+
+from .common import InfoExtractor
+
+
+class JoqrAgIE(InfoExtractor):
+    IE_DESC = '超!A&G+ 文化放送 Nippon Cultural Broadcasting, Inc. (JOQR)'
+    _VALID_URL = r'''(?x)
+                    (https?://www\.uniqueradio\.jp/agplayer5/player\.php)|
+                    (https?://www\.uniqueradio\.jp/agplayer5/inc-player-hls\.php)|
+                    (https?://(?:www\.)?joqr\.co\.jp/ag/)|
+                    (https?://(?:www\.)?joqr\.co\.jp/qr/(?:agdailyprogram|agregularprogram)/)
+                    '''
```

Comment:
you can have multiple `_VALID_URL`s without regex capture workarounds by putting them in a list like this:
```suggestion
    _VALID_URL = [r'https?://www\.uniqueradio\.jp/agplayer5/player\.php',
                  r'https?://www\.uniqueradio\.jp/agplayer5/inc-player-hls\.php',
                  r'https?://(?:www\.)?joqr\.co\.jp/ag/',
                  r'https?://(?:www\.)?joqr\.co\.jp/qr/(?:agdailyprogram|agregularprogram)/']
```

Reply:
do all of these URLs point to the same stream?

Reply:
@bashonly yes, or say, there is only one stream on this site.

Reply:
^, only the uniqueradio.jp urls actually have the stream though
the website, https://www.joqr.co.jp/ag/, and the timetables https://www.joqr.co.jp/qr/agdailyprogram/ and https://www.joqr.co.jp/qr/agregularprogram/, have some prominent links that point to the stream, but not the stream itself